File: ocaml/main2.ml
let dangerous (a:int array) (index:int) : int =
  a.(index)  (* Invalid_argument "index out of bounds" *)

let rec foo (a:int array) (counter:int) : int =
  if counter = 0 then dangerous a (counter + 9137)
  else foo a (counter - 1)

let () =
  let a = Array.make 1000 0 in
  let r = foo a 6 in
  Printf.printf "The result is %d\n" r

File: python/main2.py
def dangerous(array: list[int], index: int) -> int:
    return array[index]

def foo(array: list[int], counter: int) -> int:
    if counter == 0:
        return dangerous(array, counter + 9137)
    return foo(array, counter - 1)

def main() -> None:
    array: list[int] = [0] * 1000
    result = foo(array, 6)
    print("The result is %i", result)

if __name__ == "__main__":
    main()

